### PR TITLE
Qualitative/categorical variables

### PR DESCRIPTION
**Pull request by ralsei**
I'm making this a PR rather than just merging it because I believe it warrants careful review, especially with use-cases that I've just... missed, or combinations that break everything that I haven't found.

This PR adds support for qualitative variables in the `points` and `col` renderers. Namely, this is useful for creating dot-plots, or bar charts where you have literal data. Furthermore, it abstracts the notion of a qualitative variable altogether, as well as the notion of qualitative ticks. Unfortunately, this has resulted in a few hacks (see `gr-bars-add-ticks?` as an example of something that's probably suboptimal).

It also removes any notion of sorting variables before plotting them, and switches all hash iterations (non-deterministic) to `ddict`s. This is because this PR was originally created to add qualitative variables _and ordering_, but it is likely that ordering functionality should be delegated to a data-wrangling library, and have Graphite merely read the data in its order of appearance.

### CODE REVIEW

File: graphite-examples/socviz/5.10.rkt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+(random-seed 33) ; jittering
+
+(define organdata (df-read/csv "../data/organdata.csv" #:na "NA"))
+
+(parameterize ([plot-legend-layout '(rows 1 compact)]) ; TODO: make this part of theming
+  (graph #:data organdata
+         #:mapping (aes #:x "donors" #:y "country")
+         #:width 700 #:height 500
+         #:y-min -1 #:y-max 17 ; padding
```
Comment (samth) [inline graphite-examples/socviz/5.10.rkt:12]: Could this be part of the theme too?

File: graphite-tutorial/graphite-tutorial.scrbl [hunk 2]
```diff
@@ -284,15 +284,3 @@ represent religious preferences in that region. In that case:
 ]
 
 Now we've managed to split up our visualization into seperate charts for each region.
-
-We can do this in a similar fashion for the Gapminder plot from before, instead of using @racket[#:discrete-color]:
```
Comment (samth) [inline graphite-tutorial/graphite-tutorial.scrbl:288]: Why remove this example?

File: graphite-examples/socviz/5.10.rkt (hunk 1 above)
Comment (ralsei) [reply graphite-examples/socviz/5.10.rkt:12]: Probably?

File: graphite-tutorial/graphite-tutorial.scrbl (hunk 2 above)
Comment (ralsei) [reply graphite-tutorial/graphite-tutorial.scrbl:288]: It was broken (see #1)

**Comment by ralsei** [comment]
:pray:

**Pull request merged**